x/ibc/03-connection/keeper/handshake.go

```diff
@@ -56,68 +56,57 @@ func (k Keeper) ConnOpenTry(
 	consensusHeight uint64,
 ) error {
 	// XXX: blocked by #5078
-	/*
-		if consensusHeight > uint64(ctx.BlockHeight()) {
-			return sdkerrors.Wrap(ibctypes.ErrInvalidHeight, "invalid consensus height")
-		}
+	// if consensusHeight > uint64(ctx.BlockHeight()) {
+	// 	return sdkerrors.Wrap(ibctypes.ErrInvalidHeight, "invalid consensus height")
+	// }
 
-		expectedConsensusState, found := k.clientKeeper.GetConsensusState(ctx, clientID)
-		if !found {
-			return errors.New("client consensus state not found") // TODO: use ICS02 error
-		}
-	*/
+	expectedConsensusState, found := k.clientKeeper.GetConsensusState(ctx, clientID, consensusHeight)
+	if !found {
+		return clienttypes.ErrConsensusStateNotFound
+	}
 
-	// expectedConn defines Chain A's ConnectionEnd
+	// expectedConnection defines Chain A's ConnectionEnd
 	// NOTE: chain A's counterparty is chain B (i.e where this code is executed)
 	prefix := k.GetCommitmentPrefix()
 	expectedCounterparty := types.NewCounterparty(clientID, connectionID, prefix)
-	expectedConn := types.NewConnectionEnd(types.INIT, counterparty.ClientID, expectedCounterparty, counterpartyVersions)
+	expectedConnection := types.NewConnectionEnd(exported.INIT, counterparty.ClientID, expectedCounterparty, counterpartyVersions)
 
 	// chain B picks a version from Chain A's available versions that is compatible
 	// with the supported IBC versions
 	version := types.PickVersion(counterpartyVersions, types.GetCompatibleVersions())
 
 	// connection defines chain B's ConnectionEnd
-	connection := types.NewConnectionEnd(types.UNINITIALIZED, clientID, counterparty, []string{version})
-	expConnBz, err := k.cdc.MarshalBinaryLengthPrefixed(expectedConn)
-	if err != nil {
-		return err
-	}
+	connection := types.NewConnectionEnd(exported.UNINITIALIZED, clientID, counterparty, []string{version})
 
-	ok := k.VerifyMembership(
-		ctx, connection, proofHeight, proofInit,
-		types.ConnectionPath(counterparty.ConnectionID), expConnBz,
-	)
-	if !ok {
-		return errors.New("couldn't verify connection membership on counterparty's client") // TODO: sdk.Error
+	if err := k.VerifyConnectionState(
+		ctx, proofHeight, proofInit, counterparty.ConnectionID,
+		expectedConnection, expectedConsensusState,
+	); err != nil {
+		return err
 	}
 
 	// XXX: blocked by #5078
-	/*
-		expConsStateBz, err := k.cdc.MarshalBinaryLengthPrefixed(expectedConsensusState)
-		if err != nil {
-			return err
-		}
-
-		ok = k.VerifyMembership(
-			ctx, connection, proofHeight, proofConsensus,
-			clienttypes.ConsensusStatePath(counterparty.ClientID), expConsStateBz,
-		)
-		if !ok {
-			fmt.Sprintf("couldn't verify consensus state membership on counterparty's client\n")
-			return errors.New("couldn't verify consensus state membership on counterparty's client") // TODO: sdk.Error
-		}
-
-	*/
-
-	_, found := k.GetConnection(ctx, connectionID)
+	// err = k.VerifyClientConsensusState(
```

Comment:
What is this blocked by? https://github.com/cosmos/cosmos-sdk/issues/5078 is closed.

Reply:
see comment above